File: wrappers/java/src/test/java/com/evernym/vdrtools/wallet/ExportWalletTest.java
package com.evernym.vdrtools.wallet;

import com.evernym.vdrtools.IOException;
import com.evernym.vdrtools.IndyIntegrationTestWithSingleWallet;
import com.evernym.vdrtools.did.Did;
import org.junit.Test;

import java.io.File;
import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertTrue;


public class ExportWalletTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testExportWalletWorks() throws Exception {
		Did.createAndStoreMyDid(wallet, "{}").get();

		Wallet.exportWallet(wallet, EXPORT_CONFIG_JSON).get();

		assertTrue(new File(EXPORT_PATH).exists());
	}

	@Test
	public void testExportWalletWorksForExistsPath() throws Exception {
		assertTrue(new File(EXPORT_PATH).mkdir());

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(IOException.class));

		Wallet.exportWallet(wallet, EXPORT_CONFIG_JSON).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/ExportWalletTest.cs
﻿using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class ExportWalletTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestExportWalletWorks()
        {
            await Did.CreateAndStoreMyDidAsync(wallet, "{}");
            await wallet.ExportAsync(EXPORT_CONFIG_JSON);
            Assert.IsTrue(File.Exists(EXPORT_PATH));
            File.Delete(EXPORT_PATH);
        }

        [TestMethod]
        public async Task TestExportWalletWorksForExistsPath()
        {
            File.Create(EXPORT_PATH);

            var ex = await Assert.ThrowsExceptionAsync<IOException>(() =>
                wallet.ExportAsync(EXPORT_CONFIG_JSON)
            );
            
            File.Delete(EXPORT_PATH);
        }
    }
}
